File: app/src/main/java/com/mvc/cryptovault_android/contract/IWalletContract.java
package com.mvc.cryptovault_android.contract;

import com.mvc.cryptovault_android.base.BasePresenter;
import com.mvc.cryptovault_android.base.IBaseFragment;
import com.mvc.cryptovault_android.base.IBaseModel;
import com.mvc.cryptovault_android.bean.AllAssetBean;
import com.mvc.cryptovault_android.bean.AssetListBean;
import com.mvc.cryptovault_android.bean.MsgBean;
import com.mvc.cryptovault_android.bean.UpdateBean;

import io.reactivex.Observable;

public interface IWalletContract {
    abstract class WalletPresenter extends BasePresenter<IWalletModel, IWalletView> {
        public abstract void getAllAsset();

        public abstract void getAssetList();

        public abstract void getMsg(long timestamp, int type, int pagesize);
    }

    interface IWalletModel extends IBaseModel {
        /**
         * remove All asset
         *
         * @return
         */
        Observable<AllAssetBean> getAllAsset();

        /**
         * remove asset list
         *
         * @return
         */
        Observable<AssetListBean> getAssetList();

        /**
         * 请求数据
         *
         * @return
         */
        Observable<MsgBean> getMsg(long timestamp, int type, int pagesize);


    }

    interface IWalletView extends IBaseFragment {
        void refreshAssetList(AssetListBean asset);

        void refreshAllAsset(AllAssetBean allAssetBean);

        void refreshMsg(MsgBean msgBean);

        void showNullAsset();

        void serverError();
    }
}

File: app/src/main/java/com/mvc/ttpay_android/contract/IWalletContract.kt
package com.mvc.ttpay_android.contract

import com.mvc.ttpay_android.base.BasePresenter
import com.mvc.ttpay_android.base.IBaseFragment
import com.mvc.ttpay_android.base.IBaseModel
import com.mvc.ttpay_android.bean.*

import io.reactivex.Observable

interface IWalletContract {
    abstract class WalletPresenter : BasePresenter<IWalletModel, IWalletView>() {
        abstract fun getAllAsset()

        abstract fun getAssetList()

        abstract fun getWhetherToSignIn()

        abstract fun putSignIn()

        abstract fun getBanner()

        abstract fun getMsg(timestamp: Long, type: Int, pagesize: Int)
    }

    interface IWalletModel : IBaseModel {
        /**
         * remove All asset
         *
         * @return
         */
        val allAsset: Observable<AllAssetBean>

        /**
         * remove asset list
         *
         * @return
         */
        val assetList: Observable<AssetListBean>

        /**
         * 获取用户是否签到
         *
         * @return
         */
        val whetherToSignIn: Observable<UpdateBean>

        /**
         * 发起签到请求
         *
         * @return
         */
        fun putSignIn(): Observable<UpdateBean>

        /**
         * 发起签到请求
         *
         * @return
         */
        fun getBanner(): Observable<BannerBean>

        /**
         * 请求数据
         *
         * @return
         */
        fun getMsg(timestamp: Long, type: Int, pagesize: Int): Observable<MsgBean>


    }

    interface IWalletView : IBaseFragment {
        fun refreshAssetList(asset: AssetListBean)

        fun refreshAllAsset(allAssetBean: AllAssetBean)

        fun refreshMsg(msgBean: MsgBean)

        fun showSignin(isSignin: Boolean)

        fun signRequest(isSignin: Boolean)

        fun bannerList(bannerBean: ArrayList<BannerBean.DataBean>)

        fun showNullAsset()

        fun serverError()
    }
}
